tuna/utils/db_utility.py

```diff
@@ -68,9 +68,13 @@ def session_retry(session, callback, actuator, logger=LOGGER):
       logger.warning('%s, maybe DB contention sleeping (%s)...', error, idx)
       session.rollback()
       sleep(random.randint(1, 30))
+    except IntegrityError as error:
+      logger.error('Query failed: %s', error)
+      session.rollback()
+      return False
 
   logger.error('All retries have failed.')
-  return None
+  return False
```

Comment:
this function seems to always return False. Did you mean to return True at the end instead?